roachtest: psycopg failed

roachtest.psycopg [failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=4605086&tab=buildLog) with [artifacts](https://teamcity.cockroachdb.com/viewLog.html?buildId=4605086&tab=artifacts#/psycopg) on master @ [9850f2aed2d44e04e08d9f14ff0c98a2bd193ee9](https://github.com/cockroachdb/cockroach/commits/9850f2aed2d44e04e08d9f14ff0c98a2bd193ee9):


```
The test failed on branch=master, cloud=gce:
test artifacts and logs in: /artifacts/psycopg/run_1
	orm_helpers.go:190,orm_helpers.go:116,psycopg.go:144,psycopg.go:156,test_runner.go:866: 
		Tests run on Cockroach v22.1.0-alpha.3-320-g9850f2aed2
		Tests run against psycopg 2_8_6
		453 Total Tests Run
		451 tests passed
		2 tests failed
		319 tests skipped
		3 tests ignored
		0 tests passed unexpectedly
		1 test failed unexpectedly
		0 tests expected failed but skipped
		0 tests expected failed but not run
		---
		--- FAIL: tests.test_async.AsyncTests.test_error (unexpected)
		For a full summary look at the psycopg artifacts 
		
		An updated blocklist (psycopgBlockList22_1) is available in the artifacts' psycopg log
```
<details><summary>Help</summary>
<p>


See: [roachtest README](https://github.com/cockroachdb/cockroach/blob/master/pkg/cmd/roachtest/README.md)



See: [How To Investigate \(internal\)](https://cockroachlabs.atlassian.net/l/c/SSSBr8c7)

</p>
</details>
/cc @cockroachdb/sql-experience
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*psycopg.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>


Jira issue: CRDB-13955

error is 
```
======================================================================
FAIL: test_error (tests.test_async.AsyncTests)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "tests/test_async.py", line 414, in test_error
    self.assertEquals(cur.fetchall(), [(1, ), (2, ), (3, )])
AssertionError: Lists differ: [(1L,), (3L,)] != [(1,), (2,), (3,)]

First differing element 1:
(3L,)
(2,)

Second list contains 1 additional elements.
First extra element 2:
(3,)

- [(1L,), (3L,)]
?    -      -

+ [(1,), (2,), (3,)]
?        ++++++
```

The test is https://github.com/psycopg/psycopg2/blob/c34bf2f2f90fbbe65e8c4c2650734aa151ddd6ed/tests/test_async.py#L393-L415

It looks like there is custom CRDB logic. However, since https://github.com/cockroachdb/docs/issues/13254 has merged, i believe this test should have the custom logic removed since we match Postgres now. removing release-blocker label

roachtest.psycopg [failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=4614407&tab=buildLog) with [artifacts](https://teamcity.cockroachdb.com/viewLog.html?buildId=4614407&tab=artifacts#/psycopg) on master @ [4f361d30d788f9614695e687b386aedd783ee3ba](https://github.com/cockroachdb/cockroach/commits/4f361d30d788f9614695e687b386aedd783ee3ba):


```
The test failed on branch=master, cloud=gce:
test artifacts and logs in: /artifacts/psycopg/run_1
	orm_helpers.go:190,orm_helpers.go:116,psycopg.go:144,psycopg.go:156,test_runner.go:875: 
		Tests run on Cockroach v22.1.0-alpha.3-349-g4f361d30d7
		Tests run against psycopg 2_8_6
		453 Total Tests Run
		451 tests passed
		2 tests failed
		319 tests skipped
		3 tests ignored
		0 tests passed unexpectedly
		1 test failed unexpectedly
		0 tests expected failed but skipped
		0 tests expected failed but not run
		---
		--- FAIL: tests.test_async.AsyncTests.test_error (unexpected)
		For a full summary look at the psycopg artifacts 
		
		An updated blocklist (psycopgBlockList22_1) is available in the artifacts' psycopg log
```
<details><summary>Help</summary>
<p>


See: [roachtest README](https://github.com/cockroachdb/cockroach/blob/master/pkg/cmd/roachtest/README.md)



See: [How To Investigate \(internal\)](https://cockroachlabs.atlassian.net/l/c/SSSBr8c7)

</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*psycopg.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

roachtest.psycopg [failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=4619944&tab=buildLog) with [artifacts](https://teamcity.cockroachdb.com/viewLog.html?buildId=4619944&tab=artifacts#/psycopg) on master @ [b3525c95b176f7520173e7a6d3e4a93dca86b198](https://github.com/cockroachdb/cockroach/commits/b3525c95b176f7520173e7a6d3e4a93dca86b198):


```
The test failed on branch=master, cloud=gce:
test artifacts and logs in: /artifacts/psycopg/run_1
	orm_helpers.go:190,orm_helpers.go:116,psycopg.go:144,psycopg.go:156,test_runner.go:875: 
		Tests run on Cockroach v22.1.0-alpha.3-353-gb3525c95b1
		Tests run against psycopg 2_8_6
		453 Total Tests Run
		451 tests passed
		2 tests failed
		319 tests skipped
		3 tests ignored
		0 tests passed unexpectedly
		1 test failed unexpectedly
		0 tests expected failed but skipped
		0 tests expected failed but not run
		---
		--- FAIL: tests.test_async.AsyncTests.test_error (unexpected)
		For a full summary look at the psycopg artifacts 
		
		An updated blocklist (psycopgBlockList22_1) is available in the artifacts' psycopg log
```
<details><summary>Help</summary>
<p>


See: [roachtest README](https://github.com/cockroachdb/cockroach/blob/master/pkg/cmd/roachtest/README.md)



See: [How To Investigate \(internal\)](https://cockroachlabs.atlassian.net/l/c/SSSBr8c7)

</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*psycopg.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

roachtest.psycopg [failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=4632355&tab=buildLog) with [artifacts](https://teamcity.cockroachdb.com/viewLog.html?buildId=4632355&tab=artifacts#/psycopg) on master @ [10e0c5d92f8ef953d6b497b448893bb5044cdd31](https://github.com/cockroachdb/cockroach/commits/10e0c5d92f8ef953d6b497b448893bb5044cdd31):


```
The test failed on branch=master, cloud=gce:
test artifacts and logs in: /artifacts/psycopg/run_1
	orm_helpers.go:190,orm_helpers.go:116,psycopg.go:144,psycopg.go:156,test_runner.go:875: 
		Tests run on Cockroach v22.1.0-alpha.4-169-g10e0c5d92f
		Tests run against psycopg 2_8_6
		453 Total Tests Run
		451 tests passed
		2 tests failed
		319 tests skipped
		3 tests ignored
		0 tests passed unexpectedly
		1 test failed unexpectedly
		0 tests expected failed but skipped
		0 tests expected failed but not run
		---
		--- FAIL: tests.test_async.AsyncTests.test_error (unexpected)
		For a full summary look at the psycopg artifacts 
		
		An updated blocklist (psycopgBlockList22_1) is available in the artifacts' psycopg log
```
<details><summary>Help</summary>
<p>


See: [roachtest README](https://github.com/cockroachdb/cockroach/blob/master/pkg/cmd/roachtest/README.md)



See: [How To Investigate \(internal\)](https://cockroachlabs.atlassian.net/l/c/SSSBr8c7)

</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*psycopg.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

roachtest.psycopg [failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=4645258&tab=buildLog) with [artifacts](https://teamcity.cockroachdb.com/viewLog.html?buildId=4645258&tab=artifacts#/psycopg) on master @ [e02793d3e214c5b716149963369aa72f9338ac0d](https://github.com/cockroachdb/cockroach/commits/e02793d3e214c5b716149963369aa72f9338ac0d):


```
The test failed on branch=master, cloud=gce:
test artifacts and logs in: /artifacts/psycopg/run_1
	orm_helpers.go:190,orm_helpers.go:116,psycopg.go:144,psycopg.go:156,test_runner.go:875: 
		Tests run on Cockroach v22.1.0-alpha.4-207-ge02793d3e2
		Tests run against psycopg 2_8_6
		453 Total Tests Run
		451 tests passed
		2 tests failed
		319 tests skipped
		3 tests ignored
		0 tests passed unexpectedly
		1 test failed unexpectedly
		0 tests expected failed but skipped
		0 tests expected failed but not run
		---
		--- FAIL: tests.test_async.AsyncTests.test_error (unexpected)
		For a full summary look at the psycopg artifacts 
		
		An updated blocklist (psycopgBlockList22_1) is available in the artifacts' psycopg log
```
<details><summary>Help</summary>
<p>


See: [roachtest README](https://github.com/cockroachdb/cockroach/blob/master/pkg/cmd/roachtest/README.md)



See: [How To Investigate \(internal\)](https://cockroachlabs.atlassian.net/l/c/SSSBr8c7)

</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*psycopg.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

roachtest.psycopg [failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=4659007&tab=buildLog) with [artifacts](https://teamcity.cockroachdb.com/viewLog.html?buildId=4659007&tab=artifacts#/psycopg) on master @ [8b367174769c89c0fcfe50986ed68d4650be7750](https://github.com/cockroachdb/cockroach/commits/8b367174769c89c0fcfe50986ed68d4650be7750):


```
The test failed on branch=master, cloud=gce:
test artifacts and logs in: /artifacts/psycopg/run_1
	orm_helpers.go:190,orm_helpers.go:116,psycopg.go:144,psycopg.go:156,test_runner.go:875: 
		Tests run on Cockroach v22.1.0-alpha.4-283-g8b36717476
		Tests run against psycopg 2_8_6
		453 Total Tests Run
		451 tests passed
		2 tests failed
		319 tests skipped
		3 tests ignored
		0 tests passed unexpectedly
		1 test failed unexpectedly
		0 tests expected failed but skipped
		0 tests expected failed but not run
		---
		--- FAIL: tests.test_async.AsyncTests.test_error (unexpected)
		For a full summary look at the psycopg artifacts 
		
		An updated blocklist (psycopgBlockList22_1) is available in the artifacts' psycopg log
```
<details><summary>Help</summary>
<p>


See: [roachtest README](https://github.com/cockroachdb/cockroach/blob/master/pkg/cmd/roachtest/README.md)



See: [How To Investigate \(internal\)](https://cockroachlabs.atlassian.net/l/c/SSSBr8c7)

</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*psycopg.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

roachtest.psycopg [failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=4672349&tab=buildLog) with [artifacts](https://teamcity.cockroachdb.com/viewLog.html?buildId=4672349&tab=artifacts#/psycopg) on master @ [d2a588523715f914906f08b1365f8fc9c8ae2805](https://github.com/cockroachdb/cockroach/commits/d2a588523715f914906f08b1365f8fc9c8ae2805):


```
The test failed on branch=master, cloud=gce:
test artifacts and logs in: /artifacts/psycopg/run_1
	orm_helpers.go:190,orm_helpers.go:116,psycopg.go:144,psycopg.go:156,test_runner.go:875: 
		Tests run on Cockroach v22.1.0-alpha.4-328-gd2a5885237
		Tests run against psycopg 2_8_6
		453 Total Tests Run
		451 tests passed
		2 tests failed
		319 tests skipped
		3 tests ignored
		0 tests passed unexpectedly
		1 test failed unexpectedly
		0 tests expected failed but skipped
		0 tests expected failed but not run
		---
		--- FAIL: tests.test_async.AsyncTests.test_error (unexpected)
		For a full summary look at the psycopg artifacts 
		
		An updated blocklist (psycopgBlockList22_1) is available in the artifacts' psycopg log
```
<details><summary>Help</summary>
<p>


See: [roachtest README](https://github.com/cockroachdb/cockroach/blob/master/pkg/cmd/roachtest/README.md)



See: [How To Investigate \(internal\)](https://cockroachlabs.atlassian.net/l/c/SSSBr8c7)

</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*psycopg.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

roachtest.psycopg [failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=4683766&tab=buildLog) with [artifacts](https://teamcity.cockroachdb.com/viewLog.html?buildId=4683766&tab=artifacts#/psycopg) on master @ [090d481f2ec1d1c3e668274f50e4f152b4596a1b](https://github.com/cockroachdb/cockroach/commits/090d481f2ec1d1c3e668274f50e4f152b4596a1b):


```
The test failed on branch=master, cloud=gce:
test artifacts and logs in: /artifacts/psycopg/run_1
	orm_helpers.go:190,orm_helpers.go:116,psycopg.go:144,psycopg.go:156,test_runner.go:875: 
		Tests run on Cockroach v22.1.0-alpha.4-360-g090d481f2e
		Tests run against psycopg 2_8_6
		453 Total Tests Run
		451 tests passed
		2 tests failed
		319 tests skipped
		3 tests ignored
		0 tests passed unexpectedly
		1 test failed unexpectedly
		0 tests expected failed but skipped
		0 tests expected failed but not run
		---
		--- FAIL: tests.test_async.AsyncTests.test_error (unexpected)
		For a full summary look at the psycopg artifacts 
		
		An updated blocklist (psycopgBlockList22_1) is available in the artifacts' psycopg log
```
<details><summary>Help</summary>
<p>


See: [roachtest README](https://github.com/cockroachdb/cockroach/blob/master/pkg/cmd/roachtest/README.md)



See: [How To Investigate \(internal\)](https://cockroachlabs.atlassian.net/l/c/SSSBr8c7)

</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*psycopg.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

roachtest.psycopg [failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=4690845&tab=buildLog) with [artifacts](https://teamcity.cockroachdb.com/viewLog.html?buildId=4690845&tab=artifacts#/psycopg) on master @ [834eaa0e83350486830867b5edd6e8809b52aa55](https://github.com/cockroachdb/cockroach/commits/834eaa0e83350486830867b5edd6e8809b52aa55):


```
The test failed on branch=master, cloud=gce:
test artifacts and logs in: /artifacts/psycopg/run_1
	orm_helpers.go:190,orm_helpers.go:116,psycopg.go:144,psycopg.go:156,test_runner.go:875: 
		Tests run on Cockroach v22.1.0-alpha.4-371-g834eaa0e83
		Tests run against psycopg 2_8_6
		453 Total Tests Run
		451 tests passed
		2 tests failed
		319 tests skipped
		3 tests ignored
		0 tests passed unexpectedly
		1 test failed unexpectedly
		0 tests expected failed but skipped
		0 tests expected failed but not run
		---
		--- FAIL: tests.test_async.AsyncTests.test_error (unexpected)
		For a full summary look at the psycopg artifacts 
		
		An updated blocklist (psycopgBlockList22_1) is available in the artifacts' psycopg log
```
<details><summary>Help</summary>
<p>


See: [roachtest README](https://github.com/cockroachdb/cockroach/blob/master/pkg/cmd/roachtest/README.md)



See: [How To Investigate \(internal\)](https://cockroachlabs.atlassian.net/l/c/SSSBr8c7)

</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*psycopg.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

roachtest.psycopg [failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=4698277&tab=buildLog) with [artifacts](https://teamcity.cockroachdb.com/viewLog.html?buildId=4698277&tab=artifacts#/psycopg) on master @ [32b45c4bcf1ab41f0ba3abd36cb670eea7f450fd](https://github.com/cockroachdb/cockroach/commits/32b45c4bcf1ab41f0ba3abd36cb670eea7f450fd):


```
The test failed on branch=master, cloud=gce:
test artifacts and logs in: /artifacts/psycopg/run_1
	orm_helpers.go:190,orm_helpers.go:116,psycopg.go:144,psycopg.go:156,test_runner.go:875: 
		Tests run on Cockroach v22.1.0-alpha.4-387-g32b45c4bcf
		Tests run against psycopg 2_8_6
		453 Total Tests Run
		451 tests passed
		2 tests failed
		319 tests skipped
		3 tests ignored
		0 tests passed unexpectedly
		1 test failed unexpectedly
		0 tests expected failed but skipped
		0 tests expected failed but not run
		---
		--- FAIL: tests.test_async.AsyncTests.test_error (unexpected)
		For a full summary look at the psycopg artifacts 
		
		An updated blocklist (psycopgBlockList22_1) is available in the artifacts' psycopg log
```
<details><summary>Help</summary>
<p>


See: [roachtest README](https://github.com/cockroachdb/cockroach/blob/master/pkg/cmd/roachtest/README.md)



See: [How To Investigate \(internal\)](https://cockroachlabs.atlassian.net/l/c/SSSBr8c7)

</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*psycopg.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

--- a/pkg/cmd/roachtest/tests/psycopg.go
+++ b/pkg/cmd/roachtest/tests/psycopg.go
@@ -12,6 +12,7 @@ package tests
 
 import (
 	"context"
+	"fmt"
 	"regexp"
 
 	"github.com/cockroachdb/cockroach/pkg/cmd/roachtest/cluster"
@@ -23,7 +24,7 @@ import (
 )
 
 var psycopgReleaseTagRegex = regexp.MustCompile(`^(?P<major>\d+)(?:_(?P<minor>\d+)(?:_(?P<point>\d+)(?:_(?P<subpoint>\d+))?)?)?$`)
-var supportedPsycopgTag = "2_8_6"
+var supportedPsycopgTag = "3c58e96e1000ef60060fb8139687028cb274838d"
 
 // This test runs psycopg full test suite against a single cockroach node.
 func registerPsycopg(r registry.Registry) {
@@ -64,12 +65,9 @@ func registerPsycopg(r registry.Registry) {
 		}
 
 		if err := repeatRunE(
-			ctx,
-			t,
-			c,
-			node,
+			ctx, t, c, node,
 			"install dependencies",
-			`sudo apt-get -qq install make python3 libpq-dev python-dev gcc python3-setuptools python-setuptools`,
+			`sudo apt-get -qq install make python3 libpq-dev python3-dev gcc python3-setuptools python-setuptools`,
 		); err != nil {
 			t.Fatal(err)
 		}
@@ -80,21 +78,27 @@ func registerPsycopg(r registry.Registry) {
 			t.Fatal(err)
 		}
 
-		if err := repeatGitCloneE(
-			ctx,
-			t,
-			c,
-			"https://github.com/psycopg/psycopg2.git",
-			"/mnt/data1/psycopg",
-			supportedPsycopgTag,
-			node,
-		); err != nil {
+		// TODO(rafi): When psycopg 2.9.4 is released and tagged,
+		//    use the tag version instead of the commit.
+		//if err := repeatGitCloneE(
+		//	ctx, t, c,
+		//	"https://github.com/psycopg/psycopg2.git",
+		//	"/mnt/data1/psycopg",
+		//	supportedPsycopgTag,
+		//	node,
+		//); err != nil {
+		//	t.Fatal(err)
+		//}
+		if err = c.RunE(ctx, node, "git clone https://github.com/psycopg/psycopg2.git /mnt/data1/psycopg"); err != nil {
+			t.Fatal(err)
+		}
+		if err = c.RunE(ctx, node, fmt.Sprintf("cd /mnt/data1/psycopg/ && git checkout %s", supportedPsycopgTag)); err != nil {
 			t.Fatal(err)
 		}
 
 		t.Status("building Psycopg")
 		if err := repeatRunE(
-			ctx, t, c, node, "building Psycopg", `cd /mnt/data1/psycopg/ && make`,
+			ctx, t, c, node, "building Psycopg", `cd /mnt/data1/psycopg/ && make PYTHON_VERSION=3`,
 		); err != nil {
 			t.Fatal(err)
 		}
@@ -117,7 +121,7 @@ func registerPsycopg(r registry.Registry) {
 			export PSYCOPG2_TESTDB_USER=root &&
 			export PSYCOPG2_TESTDB_PORT=26257 &&
 			export PSYCOPG2_TESTDB_HOST=localhost &&
-			make check`,
+			make check PYTHON_VERSION=3`,
 		)
 
 		// Expected to fail but we should still scan the error to check if

--- a/pkg/cmd/roachtest/tests/psycopg_blocklist.go
+++ b/pkg/cmd/roachtest/tests/psycopg_blocklist.go
@@ -28,10 +28,7 @@ var psycopgBlocklists = blocklistsForVersion{
 // Please keep these lists alphabetized for easy diffing.
 // After a failed run, an updated version of this blocklist should be available
 // in the test log.
-var psycopgBlockList22_1 = blocklist{
-	// The following item can be removed once there is a new psycopg2 release.
-	"tests.test_module.ExceptionsTestCase.test_9_6_diagnostics": "58035",
-}
+var psycopgBlockList22_1 = blocklist{}
 
 var psycopgBlockList21_2 = blocklist{
 	"tests.test_async_keyword.CancelTests.test_async_cancel": "41335",

